doc/data/messages/c/consider-using-assignment-expr/pylintrc

```diff
@@ -0,0 +1,2 @@
+[MAIN]
+py-version=3.8
```

Comment:
@Pierre-Sassoulas would this not have been sufficient to make the CI check `Checks / documentation` pass?

Reply:
Does ``pylint /home/runner/work/pylint/pylint/doc/data/messages/c/consider-using-assignment-expr/bad.py --disable=all --enable=F,consider-using-assignment-expr,astroid-error,syntax-error`` work locally ?

Reply:
There might be an issue with the decorator "only used for" on the check it happened often.

Reply:
It indeed does not work locally, No `consider-using-assignment-expr` is in the output. Is there something you suggest to do in this case to make this work?

Reply:
I would start by checking the ``util.only_required_for_messages`` calls when ``consider-using-assignment-expr`` is involved in the decorated function. Oftentimes this is the reason a check do not work if other checks are disabled.

Reply:
This is a function used to optimize execution time to not run code when not needed.

Reply:
I think the issue is in using this pylintrc file  inside the `.../c/consider-using-assignment-expr` dir. Take a look:

```
 pylint doc/data/messages/c/consider-using-assignment-expr/bad.py --disable=all --enable=F,consider-using-assignment-expr,astroid-error,syntax-error --py-version 3.8
************* Module bad
doc/data/messages/c/consider-using-assignment-expr/bad.py:3:3: R6103: Use 'if (apples := 2):' instead (consider-using-assignment-expr)
```

passing py-version as an arg works as expected! I assumed since other message dirs have a pylintrc file it should work?

Reply:
It's possible that py-version is not handled in the functional test configuration. py-version was introduced after it and functional test helpers are not super well tested.

Reply:
I assumed loading pylintrc configs wasn't the responsibility of the functional tests in test_message_documentation.py, but instead is a core pylint responsibility. Am I mistaken?

Reply:
Interesting, unexpected results:

```
pylint doc/data/messages/c/consider-using-assignment-expr/bad.py --disable=all --enable=F,consider-using-assignment-expr,astroid-error,syntax-error --py-version 3.8
************* Module bad
doc/data/messages/c/consider-using-assignment-expr/bad.py:3:3: R6103: Use 'if (apples := 2):' instead (consider-using-assignment-expr)
```

with pylintrc contents as in this PR
```
[MAIN]
py-version=3.8
```

we get diff output
```
pylint doc/data/messages/c/consider-using-assignment-expr/bad.py --disable=all --enable=F,consider-using-assignment-expr,astroid-error,syntax-error --rcfile=doc/data/messages/c/consider-using-assignment-expr/pylintrc

-------------------------------------------------------------------
```

This seems fishy to me...

Reply:
Yes indeed, if ``py-version=3.8`` then the 3.8 checks should be activated not the other way around.

Reply:
It's worth noting `consider-using-assignment-exp` is not within the `extensions` dir... but I see it mentioned in extensions.rst 🤷‍♀️